src/naive_bayes/categorical.rs

```diff
@@ -6,14 +6,13 @@ use crate::naive_bayes::{BaseNaiveBayes, NBDistribution};
 use serde::{Deserialize, Serialize};
 
 /// Naive Bayes classifier for categorical features
-#[derive(Debug)]
-struct CategoricalNBDistribution<T: RealNumber> {
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+struct CategoricalNBDistribution<T: RealNumber, M: Matrix<T>> {
     class_labels: Vec<T>,
     class_priors: Vec<T>,
-    coefficients: Vec<Vec<Vec<T>>>,
+    coefficients: Vec<Vec<M>>,
```

Comment:
Can you use M::RowVector here instead? This will let you avoid code like this:
```
let coef_i_j = M::RowVector::from_array(&coef_i_j);
let coef_i_j = M::from_row_vector(coef_i_j);
```

Reply:
The problem with that is that RowVector does not have a PartialEq implementation, let me check if I can do that

Reply:
I think we should add `+ PartialEq` to BaseVector

Reply:
Another associated problem is this:

```rust
13 |     coefficients: Vec<Vec<M::RowVector>>,
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `error::_::_serde::Deserialize<'_>` is not implemented for `<M as linalg::BaseMatrix<T>>::RowVector`
   |
```

Reply:
I would have to implement a custom way to serialize and deserialize CategoricalNBDistribution, right?

Reply:
I've seen it before. [This solution](https://github.com/smartcorelib/smartcore/blob/development/src/svm/svc.rs#L106) seems to help

Reply:
On the other hand, you don't have to go through the trouble of storing coefficients as `Vec<Vec<M::RowVector>>`. You can keep using `Vec<Vec<Vec<T>>>` as you did before and define custom `PartialEq` for your `CategoricalNBDistribution` that has a method `eq`
```
impl<T: RealNumber> PartialEq for KMeans<T> {
    fn eq(&self, other: &Self) -> bool {
        ...
        for i in 0..self.coefficients.len() {
               for j in 0..self.coefficients[j] {
                    for k in 0..self.coefficients[j][k] {
                        if (self.coefficients[i][j][k] - other.centroids[i][j][k]).abs() > T::epsilon() {
                           return false;
                        }
                }
         ...
    }
}
```

Reply:
I applied your suggestion and it worked fine but the tests are still failing due to the default PartialEq implementation of Vec<_>. I am going to follow your suggestion and use a custom implementation of PartialEq for CategoricalNBDistribution as I previously did in https://github.com/smartcorelib/smartcore/pull/30/commits/a23e66c5dc15ef0d604afbac6dee21f27f69feca